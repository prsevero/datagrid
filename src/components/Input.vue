<template>
    <label>
        <input :class="{'input-error': hasError}" :required="required" :type="type" :value="value" @input="$emit('input', $event.target.value)">
        <span>{{ label }}</span>
    </label>
</template>


<script>
export default {
    name: 'Input',
    props: {
        hasError: Boolean,
        label: {
            required: true,
            type: String,
        },
        required: Boolean,
        type: {
            required: true,
            type: String,
        },
        value: [String, Number],
    },
};
</script>

<style lang="sass" scoped>
label
    display: block
    margin: 0 0 20px
    position: relative
    span
        background: #fff
        color: #333
        cursor: text
        left: 5px
        padding: 0 4px
        pointer-events: none
        position: absolute
        top: 12px
        transition: all .15s ease-in-out

input
    appearance: none
    border: 1px solid #ccc
    border-radius: 0
    color: #333
    display: block
    outline: 0
    padding: 8px
    width: 100%
    transition: all .22s ease-in-out
    &.input-error
        border-color: #f00
    &:focus
        border-color: #777
    &:focus, &:valid
        ~ span
            color: #444
            font-size: .8em
            top: -9px
</style>
