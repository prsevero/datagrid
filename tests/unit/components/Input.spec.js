import { shallowMount } from '@vue/test-utils';
import Input from '@/components/Input.vue';

describe('Input.vue', () => {

    let props;
    let mountedInput;
    const input = () => {
        if (!mountedInput) {
            mountedInput = shallowMount(Input, {propsData: props});
        }
        return mountedInput;
    }

    beforeEach(() => {
        props = {
            label: '',
            type: '',
        };
        mountedInput = undefined;
    });


    // All tests will go here
    it('should match snapshot', () => {
        const wrapper = input();
        expect(wrapper.vm.$el).toMatchSnapshot();
    });

    it('always render an input and a span inside a label', () => {
        const wrapper = input();
        expect(wrapper.is('label')).toBe(true);
        expect(wrapper.findAll('input').length).toBe(1);
        expect(wrapper.findAll('span').length).toBe(1);
    });

    it('should match the label prop to the span text', () => {
        const label = 'Label';
        props.label = label;
        const wrapper = input();
        expect(wrapper.find('span').text()).toBe(label);
    });

    it('should match the label prop to the span text', () => {
        const type = 'text';
        props.type = type;
        const wrapper = input();
        expect(wrapper.find('input').element.type).toBe(type);
    });

    it('should match the value prop to the input value', () => {
        const value = 'value';
        props.value = value;
        const wrapper = input();
        expect(wrapper.find('input').element.value).toBe(value);
    });

    it('should be required if the required prop is true', () => {
        props.required = true;
        const wrapper = input();
        expect(wrapper.find('input').element.required).toBeTruthy();
    });

    it('should has error class if hasError prop is true', () => {
        props.hasError = true;
        const wrapper = input();
        expect(wrapper.find('input').element.className).toBe('input-error');
    });

});
