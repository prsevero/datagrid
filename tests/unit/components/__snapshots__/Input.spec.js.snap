// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.vue should match snapshot 1`] = `
<label>
  <input
    class=""
    type=""
  />
   
  <span>
    
  </span>
</label>
`;
